velox/functions/prestosql/aggregates/tests/PrestoHasherTest.cpp

```diff
@@ -0,0 +1,371 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <gtest/gtest.h>
+
+#include "velox/exec/tests/utils/VectorTestBase.h"
+#include "velox/functions/prestosql/aggregates/PrestoHasher.h"
+
+namespace facebook::velox::aggregate::test {
+template <typename T>
+using limits = std::numeric_limits<T>;
+
+class PrestoHasherTest : public testing::Test,
+                         public exec::test::VectorTestBase {
+ protected:
+  template <typename T>
+  void assertHash(
+      const std::vector<std::optional<T>>& data,
+      const std::vector<int64_t>& expected) {
+    auto vector = makeNullableFlatVector<T>(data);
+    assertHash(vector, expected);
+  }
+
+  void assertHash(
+      const VectorPtr& vector,
+      const std::vector<int64_t>& expected) {
+    auto vectorSize = vector->size();
+
+    auto checkHashes = [&](const VectorPtr& vector,
+                           const std::vector<int64_t>& expected) {
+      auto vectorSize = vector->size();
+      SelectivityVector rows(vectorSize);
+      PrestoHasher hasher(vector->type());
+
+      BufferPtr hashes =
+          AlignedBuffer::allocate<int64_t>(vectorSize, pool_.get());
+      hasher.hash(vector, rows, hashes);
+
+      auto rawHashes = hashes->as<int64_t>();
+      for (int i = 0; i < expected.size(); i++) {
+        EXPECT_EQ(expected[i], rawHashes[i]) << "at " << i;
+      }
+
+      // Check for a subset of rows.
+      SelectivityVector someRows(vectorSize, false);
+      for (int i = 0; i < vectorSize; i += 2) {
+        someRows.setValid(i, true);
+      }
+      someRows.updateBounds();
+      PrestoHasher someHash(vector->type());
```

Comment:
It would be nice to reuse the "hasher" instance to provide more coverage for the common use case of re-using the PrestoHasher instances.